File: pathfinder-api/src/main/java/de/cubbossa/pathapi/storage/cache/VisualizerTypeCache.java
package de.cubbossa.pathapi.storage.cache;

import de.cubbossa.pathapi.misc.NamespacedKey;
import de.cubbossa.pathapi.visualizer.PathVisualizer;
import de.cubbossa.pathapi.visualizer.VisualizerType;

import java.util.Collection;
import java.util.Map;
import java.util.Optional;

public interface VisualizerTypeCache extends StorageCache<Map.Entry<NamespacedKey, VisualizerType<?>>> {

  CacheMap<NamespacedKey, VisualizerType<?>> getTypes(Collection<NamespacedKey> key);

  <VisualizerT extends PathVisualizer<?, ?>> Optional<VisualizerType<VisualizerT>> getType(NamespacedKey key);

  <VisualizerT extends PathVisualizer<?, ?>> void write(NamespacedKey key, VisualizerType<VisualizerT> type);
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/cache/VisualizerTypeCache.kt
package de.cubbossa.pathfinder.storage.cache

import de.cubbossa.pathfinder.misc.NamespacedKey
import de.cubbossa.pathfinder.visualizer.PathVisualizer
import de.cubbossa.pathfinder.visualizer.VisualizerType

interface VisualizerTypeCache : StorageCache<Map.Entry<NamespacedKey, VisualizerType<*>>> {
    fun getTypes(key: Collection<NamespacedKey>): StorageCache.CacheMap<NamespacedKey, VisualizerType<*>>

    fun <VisualizerT : PathVisualizer<*, *>> getType(key: NamespacedKey): VisualizerType<VisualizerT>?

    fun <VisualizerT : PathVisualizer<*, *>> write(
        key: NamespacedKey,
        type: VisualizerType<VisualizerT>
    )
}
